common/flatpak-dir.c

```diff
@@ -198,6 +198,193 @@ get_config_dir_location (void)
   return (const char *)path;
 }
 
+static FlatpakRemoteState *
+flatpak_remote_state_new (void)
+{
+  return g_new0 (FlatpakRemoteState, 1);
+}
+
+void
+flatpak_remote_state_free (FlatpakRemoteState *remote_state)
+{
+  g_free (remote_state->remote);
+  g_free (remote_state->collection_id);
+  g_clear_pointer (&remote_state->summary, g_variant_unref);
+  g_clear_pointer (&remote_state->summary_sig_bytes, g_bytes_unref);
+  g_clear_pointer (&remote_state->metadata, g_variant_unref);
+
+  g_free (remote_state);
+}
+
+gboolean
+flatpak_remote_state_ensure_summary (FlatpakRemoteState *self,
+                                     GError      **error)
+{
+  if (self->summary == NULL)
+    return flatpak_fail (error, "Unable to load summary from remote %s", self->remote);
+
+  return TRUE;
+}
+
+gboolean
+flatpak_remote_state_ensure_metadata (FlatpakRemoteState *self,
+                                      GError      **error)
+{
+  if (self->metadata == NULL)
+    return flatpak_fail (error, "Unable to load medata from remote %s", self->remote);
+
+  return TRUE;
+}
+
+
+char *
+flatpak_remote_state_lookup_ref (FlatpakRemoteState *self,
+                                 const   char *ref,
+                                 GVariant    **out_variant,
+                                 GError      **error)
+{
+  g_autofree char *latest_rev = NULL;
+
+  if (!flatpak_remote_state_ensure_summary (self, error))
+    return NULL;
+
+  if (!flatpak_summary_lookup_ref (self->summary, self->collection_id, ref, &latest_rev, out_variant))
+    {
+      if (self->collection_id != NULL)
+        flatpak_fail (error, "No such ref (%s, %s) in remote %s", self->collection_id, ref, self->remote);
+      else
+        flatpak_fail (error, "No such ref '%s' in remote %s", ref, self->remote);
+      return NULL;
+    }
+
+  return g_steal_pointer (&latest_rev);
+}
+
+char **
+flatpak_remote_state_match_subrefs (FlatpakRemoteState *self,
+                                    const   char *ref)
```

Comment:
Nitpick: It would be nice to have some gtk-doc comments; even if this isn’t public API, it’s useful for developers.

Reply:
For example, the documentation here should probably say that you should typically call `ensure_summary()` first.